components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java

```diff
@@ -0,0 +1,145 @@
+import org.wso2.carbon.identity.input.validation.mgt.model.Validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.MAX_LENGTH;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.MIN_LENGTH;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_DEFAULT_MIN_MAX_MISMATCH;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_PROPERTY_NOT_SUPPORTED;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_PROPERTY_TYPE_MISMATCH;
+
+/**
+ * Abstract rules validator.
+ */
+public abstract class AbstractRulesValidator implements Validator {
+
+    private static final Log log = LogFactory.getLog(AbstractRulesValidator.class);
+
+    @Override
+    public boolean canHandle(String validatorName) {
+
+        return StringUtils.equalsIgnoreCase(validatorName, this.getClass().getSimpleName());
+    }
+
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        return true;
+    }
+
+    @Override
+    public boolean validateProps(ValidationContext context) throws InputValidationMgtClientException {
+
+        Map<String, String> properties = context.getProperties();
+        // Validated supported properties.
+        validatePropertyName(properties, this.getClass().getSimpleName(), context.getTenantDomain());
+
+        if (properties.get(MIN_LENGTH) != null && validatePositiveNumber(properties.get(MIN_LENGTH), MIN_LENGTH,
+                context.getTenantDomain())) {
+            properties.remove(MIN_LENGTH);
+        }
+        if (properties.get(MAX_LENGTH) != null && validatePositiveNumber(properties.get(MAX_LENGTH), MAX_LENGTH,
+                context.getTenantDomain())) {
+            properties.remove(MIN_LENGTH);
+        }
+        // Ensure maximum limit is not less than minimum limit.
+        if ((properties.get(MAX_LENGTH) != null) && (properties.get(MIN_LENGTH) != null) &&
+                Integer.parseInt(properties.get(MIN_LENGTH)) > Integer.parseInt(properties.get(MAX_LENGTH))) {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("The min limit cannot be greater than max limit for %s in tenant: %s",
+                        this.getClass().getSimpleName(), context.getTenantDomain()));
+            }
+            throw new InputValidationMgtClientException(ERROR_DEFAULT_MIN_MAX_MISMATCH.getCode(),
+                    String.format(ERROR_DEFAULT_MIN_MAX_MISMATCH.getDescription(), this.getClass().getSimpleName(),
+                            properties.get(MIN_LENGTH), properties.get(MAX_LENGTH)));
+        }
+        return true;
+    }
+
+
+    @Override
+    public List<Property> getConfigurationProperties() {
```

Comment:
Here, this method gets executed everytime we retrieve config properties. This does not dynamically change. can't we do this in validator creation. Keep list of properties as an attribute of the validator.

Reply:
We are not creating validators in run time. The validators are registered initially. This method does not have any db calls. I feel like there is no harm in calling this hardcoded method.